core/base/trackingFromCriticalPoints/TrackingFromCriticalPoints.cpp

```diff
@@ -301,6 +328,14 @@ void ttk::TrackingFromCriticalPoints::performTrackingForOneType(
         trackingCosts[trackingId].push_back(std::get<2>(matchings[i][j]));
         trackingPersistence[trackingId]
           += persistenceDiagrams[i + 1][map[i + 1][endLocalId]].persistence();
+        double newMax = std::max(
```

Comment:
We need to discuss this piece of code.
It looks much more complicated than it needs to.